### PR TITLE
Provider VDC without network pool

### PR DESCRIPTION
**Pull request by dataclouder**
Address Issue #1186.
This change allows Provider VDC to be created without a network pool.

Fixes Issue #1221

### CODE REVIEW

File: .changes/v3.12.0/1120-bug-fixes.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Fix [Issue #1120](https://github.com/vmware/terraform-provider-vcd/issues/1221) Portgroup backed network pool can't have a data source [GH-1220]
```
Comment (adambarreiro) [inline .changes/v3.12.0/1120-bug-fixes.md:1]:
```suggestion
Fix [Issue #1121](https://github.com/vmware/terraform-provider-vcd/issues/1221) Portgroup backed network pool can't have a data source [GH-1220]
```

File: vcd/resource_vcd_provider_vdc.go [hunk 2]
```diff
@@ -128,7 +128,7 @@ func resourceVcdProviderVdc() *schema.Resource {
 			},
 			"nsxt_manager_id": {
 				Type:        schema.TypeString,
-				Required:    true,
+				Optional:    true,
```
Comment (adambarreiro) [inline vcd/resource_vcd_provider_vdc.go:131]: Not sure if needed at this point, but just noting that this maybe needs an entry at `skip-upgrade-tests.txt`?

**Review by adambarreiro — APPROVED** [review]

**Review by lvirbalas — APPROVED** [review]

File: vcd/resource_vcd_provider_vdc.go (hunk 2 above)
Comment (dataclouder) [reply vcd/resource_vcd_provider_vdc.go:131]: It should not be needed. What worked in previous version as "Required" will work in this version as "Optional"

File: .changes/v3.12.0/1120-bug-fixes.md (hunk 1 above)
Comment (dataclouder) [reply .changes/v3.12.0/1120-bug-fixes.md]: Thanks. Fixed

**Review by Didainius — APPROVED** [review]

**Pull request merged**